ros2_dsrc_driver/src/ros2_dsrc_driver_node.cpp

```diff
@@ -0,0 +1,403 @@
+                         "      },\n"
+                         "      \"dsrc_id\": {\n"
+                         "        \"description\": \"J2735 DSRC id assigned to message type in decimal\",\n"
+                         "        \"type\": \"string\"\n"
+                         "      },\n"
+                         "      \"channel\": {\n"
+                         "        \"description\": \"DSRC radio channel assigned to message type in decimal\",\n"
+                         "        \"type\": \"string\"\n"
+                         "      },\n"
+                         "      \"priority\": {\n"
+                         "        \"description\": \"WSM Priotiy to use assigned to message type in decimal\",\n"
+                         "        \"type\":\"string\"\n"
+                         "      }\n"
+                         "    },\n"
+                         "    \"required\":[\"name\",\"psid\",\"dsrc_id\",\"channel\",\"priority\"]"
+                         "  }\n"
+                         "}\n";
+
+    std::ifstream file;
+    try
+    {
+        file.open(fileName);
+    }
+    catch (std::exception& e)
+    {
+        RCLCPP_ERROR_STREAM(this->get_logger(),"Unable to open file : " << fileName << ", exception: " << e.what());
+        return;
+    }
+
+    rapidjson::Document sd;
+    if(sd.Parse(schema).HasParseError())
+    {
+        RCLCPP_ERROR_STREAM(this->get_logger(),"Invalid Wave Config Schema");
+        return;
+    }
+
+    rapidjson::SchemaDocument schemaDocument(sd);
+    rapidjson::Document doc;
+    rapidjson::IStreamWrapper isw(file);
+    if(doc.ParseStream(isw).HasParseError())
+    {
+        RCLCPP_ERROR_STREAM(this->get_logger(),"Error Parsing Wave Config");
+        return;
+    }
+
+    rapidjson::SchemaValidator validator(schemaDocument);
+    if(!doc.Accept(validator))
+    {
+        RCLCPP_ERROR_STREAM(this->get_logger(),"Wave Config improperly formatted");
+        return;
+    }
+
+    for(auto& it : doc.GetArray())
+    {
+        auto entry = it.GetObject();
+        wave_cfg_items_.emplace_back(entry["name"].GetString(),
+                                     entry["psid"].GetString(),
+                                     entry["dsrc_id"].GetString(),
+                                     entry["channel"].GetString(),
+                                     entry["priority"].GetString());
+
+    }
+}
+
+void Node::handle_on_shutdown()
+{
+    RCLCPP_INFO_STREAM(this->get_logger(),"Shutdown signal received from DriverApplication");
+    if (connect_thread_)
+    {
+        RCLCPP_INFO_STREAM(this->get_logger(),"Cleaning up connection thread");
+        connect_thread_->join();
+        connect_thread_.reset();
+    }
+
+    RCLCPP_INFO_STREAM(this->get_logger(),"Closing connection to radio");
+    dsrc_client_.close();
+}
+
+}
+
+#include "rclcpp_components/register_node_macro.hpp"
+
+// Register the component with class_loader
+RCLCPP_COMPONENTS_REGISTER_NODE(DSRCApplication::Node)
```

Comment:
The registered node should be the same as that in CMake "ros2_dsrc_driver::Node" Since the namespace is called DSRCApplication, it would probably make more sense to update CMakeLists

Reply:
please check the cmake gain